src/gt4py/ir/nodes.py

```diff
@@ -421,11 +423,65 @@ class NativeFunction(enum.Enum):
     CEIL = 112
     TRUNC = 113
 
+    @property
+    def numargs(self):
+        return type(self).IR_OP_TO_NUM_ARGS[self]
+
+    @property
+    def python_symbol(self):
+        return type(self).IR_OP_TO_PYTHON_SYMBOL[self]
+
+
+NativeFunction.IR_OP_TO_NUM_ARGS = {
+    NativeFunction.ABS: 1,
+    NativeFunction.MIN: 2,
+    NativeFunction.MAX: 2,
+    NativeFunction.MOD: 2,
+    NativeFunction.SIN: 1,
+    NativeFunction.COS: 1,
+    NativeFunction.TAN: 1,
+    NativeFunction.ARCSIN: 1,
+    NativeFunction.ARCCOS: 1,
+    NativeFunction.ARCTAN: 1,
+    NativeFunction.SQRT: 1,
+    NativeFunction.EXP: 1,
+    NativeFunction.LOG: 1,
+    NativeFunction.ISFINITE: 1,
+    NativeFunction.ISINF: 1,
+    NativeFunction.ISNAN: 1,
+    NativeFunction.FLOOR: 1,
+    NativeFunction.CEIL: 1,
+    NativeFunction.TRUNC: 1,
+}
+
+NativeFunction.IR_OP_TO_PYTHON_SYMBOL = {
+    NativeFunction.ABS: "abs",
+    NativeFunction.MIN: "min",
+    NativeFunction.MAX: "max",
+    NativeFunction.MOD: "mod",
+    NativeFunction.SIN: "sin",
+    NativeFunction.COS: "cos",
+    NativeFunction.TAN: "tan",
+    NativeFunction.ARCSIN: "arcsin",
```

Comment:
in our discussion we decided it feels more natural the have the more standard name in the IR (since the IR is not very python related but more so a way to describe computational patterns). The place where python actually comes in is in the code-generation where we map IR concepts to python functions.